worker/uniter/metrics.go

```diff
@@ -45,11 +51,22 @@ func (t *timerChooser) getMetricsTimer(ch corecharm.Charm) TimedSignal {
 	return t.inactive
 }
 
+// getSendMetricsTimer returns the metrics timer we should be using, given the supplied
```

Comment:
Feels like we should rename `getMetricsTimer` to `getCollectMetricsTimer`. And in general, if it was assumed before that an entity named "-Metrics" was for the collect-metrics hook, now we should make that naming explicit, since collecting and sending metrics are now decoupled in this PR.